paimon-core/src/main/java/org/apache/paimon/utils/SnapshotManager.java

```diff
@@ -364,6 +365,20 @@ public Iterator<Snapshot> snapshots() throws IOException {
                 .iterator();
     }
 
+    public Iterator<Snapshot> snapshotsWithinRange(Long maxSnapshotId, Long minSnapshotId)
+            throws IOException {
+        // check for specific snapshot
+        if (Objects.equals(maxSnapshotId, minSnapshotId)) {
+            return Collections.singletonList(snapshot(maxSnapshotId)).iterator();
+        }
+
+        return listVersionedFiles(fileIO, snapshotDirectory(), SNAPSHOT_PREFIX)
```

Comment:
We still need to list directory.
Can we directly read one by one?

We can combine maxSnapshotId, minSnapshotId, earliest and latest. Calculate a queryable interval and directly call the `snapshot` method one by one.

Reply:
Removed list call, lmk what you think